### DISCUSSION TITLE
libcharon on iOS

### DISCUSSION THREAD

**Discussion by ahmedkhan1039** [strongswan/strongswan#1789] [General]
Hi @tobiasbrunner,

I have a use case where i want to have multiple simultaneous connections on an iOS application. Is it possible to use `libcharon` on iOS. I have seen it is compatible with Android but will it work on iOS ?

Thanks

**Comment by tobiasbrunner** [comment]
Technically that's possible (there are people who do that). However, we currently don't provide the `libandroidbridge` equivalent for iOS (that would interact with Apple's Network Extensions and the GUI etc.) or any detailed build instructions (to some degree the ones for the old macOS app might be useful, see [`src/frontends/osx/README.md`](https://github.com/strongswan/strongswan/blob/master/src/frontends/osx/README.md)).

**Reply by ahmedkhan1039** [reply]
Thanks for your reply.

**Reply by ahmedkhan1039** [reply]
One more thing, @tobiasbrunner 
Is it possible to have multiple simultaneous active ipsec connections on Android or iOS.? Because on Android, i have seen the implementation supports only one active connection at a time.

**Reply by tobiasbrunner** [reply]
Technically, yes. But the Android GUI and `libandroidbridge` currently only support managing a single IKE and Child SA.
